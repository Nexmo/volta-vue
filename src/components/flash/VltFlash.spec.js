import { shallowMount } from '@vue/test-utils';
import { VltFlash } from '..';

describe('vlt-flash', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(VltFlash);
  });

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('is type critical', () => {
    wrapper.setProps({ type: 'critical' });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('is visible', () => {
    wrapper.setProps({ visible: true });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('is dismissable', async () => {
    wrapper.setProps({ dismissable: true });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('can be dismissed', async () => {
    wrapper.setProps({ dismissable: true });
    wrapper.find('button').trigger('click');
    expect(wrapper.emitted('dismissed')).toBeTruthy();
  });

  test('is a bottom flash', async () => {
    wrapper.setProps({ bottom: true });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('is a small flash', async () => {
    wrapper.setProps({ small: true });
    expect(wrapper.element).toMatchSnapshot();
  });
});
