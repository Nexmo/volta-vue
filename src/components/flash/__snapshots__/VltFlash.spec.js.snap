// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vlt-flash is a bottom flash 1`] = `
<div
  class="Vlt-callout Vlt-flash Vlt-callout--shoutout Vlt-flash--bottom"
>
  <i />
  <div
    class="Vlt-callout__content"
  >
    <p>
      
      
    </p>
  </div>
  <!--v-if-->
</div>
`;

exports[`vlt-flash is a small flash 1`] = `
<div
  class="Vlt-callout Vlt-flash Vlt-callout--shoutout Vlt-flash--small"
>
  <i />
  <div
    class="Vlt-callout__content"
  >
    <p>
      
      
    </p>
  </div>
  <!--v-if-->
</div>
`;

exports[`vlt-flash is dismissable 1`] = `
<div
  class="Vlt-callout Vlt-flash Vlt-callout--shoutout"
>
  <i />
  <div
    class="Vlt-callout__content"
  >
    <p>
      
      
    </p>
  </div>
  <button
    class="Vlt-callout__dismiss"
  />
</div>
`;

exports[`vlt-flash is type critical 1`] = `
<div
  class="Vlt-callout Vlt-flash Vlt-callout--critical"
>
  <i />
  <div
    class="Vlt-callout__content"
  >
    <p>
      
      
    </p>
  </div>
  <!--v-if-->
</div>
`;

exports[`vlt-flash is visible 1`] = `
<div
  class="Vlt-callout Vlt-flash Vlt-callout--shoutout Vlt-flash_visible"
  style="top: 20px;"
>
  <i />
  <div
    class="Vlt-callout__content"
  >
    <p>
      
      
    </p>
  </div>
  <!--v-if-->
</div>
`;

exports[`vlt-flash renders correctly 1`] = `
<div
  class="Vlt-callout Vlt-flash Vlt-callout--shoutout"
>
  <i />
  <div
    class="Vlt-callout__content"
  >
    <p>
      
      
    </p>
  </div>
  <!--v-if-->
</div>
`;
